### DISCUSSION TITLE
Reason for Dropping stores

### DISCUSSION THREAD

**Discussion by SruthyJohn** [google/or-tools#3305] [Routing (and legacy CP) questions] [answered]
Hi Team,
I am doing a PoC with Google OR tools.
I have added different constraints like Vehicle capacity, Maximum order count for each store, service times,etc and I am able get solution for these cases.
I have added Disjunction to make all nodes optional, so if there is any violated constraints stores will be dropped to find a solution. I am logging these dropped stores also for the user to view those stores. But how will I get to know why a store has been dropped from the solution?

Gist here: https://gist.github.com/SruthyJohn/bd6999a9eda837f81c096fc940fbc69e.js

**Comment by lperron** [comment, marked as answer]
The solver does not/cannot report why nodes were dropped.

**Comment by jmarca** [comment]
One approach to identifying possible reasons a node might be dropped
that I use is to print out the dropped nodes along with the most
obvious reasons why it might be dropped.

For example (and if it makes sense),

* distance/time to depot
* distance/time to nearest node
* time window at node
* demand at node

Usually after printing these sorts of things, it is easy to spot why a
node is dropped.  For example, the demand might be large, and the
solver can fit multiple other nodes instead of the single dropped
node, thus minimizing the dropped node penalties.

And sometimes I find bugs this way, such as a mis-coded
opening/closing time or an incorrect (lat, lon) pair creating an
incorrect distance number.

Finally, your gist link seems messed up.  I deleted the ".js" at the end
and it worked okay.  You might want to edit your post.

Regards,

James

--

James E. Marca
Activimetrics LLC
